template/src/utils/customQueries.js

```diff
@@ -0,0 +1,37 @@
+import { fetchBaseQuery } from '@reduxjs/toolkit/dist/query/react';
+
+import { selectAuth } from 'services/auth';
+import { ACCESS_TOKEN, APPLICATION_JSON, CLIENT, CONTENT_TYPE, UID } from 'constants/headers';
+
+const baseQuery = fetchBaseQuery({
+  baseUrl: process.env.REACT_APP_API_URL,
+  prepareHeaders: (headers, { getState }) => {
+    const { info } = selectAuth(getState());
+
+    if (info.token) {
+      headers.set(CONTENT_TYPE, APPLICATION_JSON);
+      headers.set(ACCESS_TOKEN, info.token);
+      headers.set(CLIENT, info.client);
+      headers.set(UID, info.uid);
+    }
+    return headers;
+  },
+});
+
+export const baseQueryWithReauth = async (args, api, extraOptions) => {
+  const result = await baseQuery(args, api, extraOptions);
+  const headers = result?.meta?.response?.headers;
+  const token = headers.get(ACCESS_TOKEN);
```

Comment:
shouldn't be

```suggestion
  const headers = result?.meta?.response?.headers;
  const token = headers?.get(ACCESS_TOKEN);
```
too ? since we have the checks to get headers

Reply:
You're right!